spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -147,13 +156,23 @@ public SparkTable createTable(Identifier ident, StructType schema,
                                 Transform[] transforms,
                                 Map<String, String> properties) throws TableAlreadyExistsException {
     Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    TableIdentifier tableIdentifier = buildIdentifier(ident);
     try {
-      Table icebergTable = icebergCatalog.createTable(
-          buildIdentifier(ident),
-          icebergSchema,
-          Spark3Util.toPartitionSpec(icebergSchema, transforms),
-          properties.get("location"),
-          Spark3Util.rebuildCreateProperties(properties));
+      Table icebergTable;
+      if (isPathIdentifier(ident)) {
```

Comment:
I think this would be cleaner using the builder. The only problem there is that the builder isn't implemented for `CachingCatalog` yet. I'll open a PR with that builder. Then you'd be able to have an if to create the builder, then configure the builder the same way for spec, properties, identifier, etc.

Reply:
Opened https://github.com/apache/iceberg/pull/1879 with this.